### ISSUE TITLE
[Julia] `visit_Cast` does not recognize arrays properly

### ISSUE THREAD

**Issue by lucifer1004** [lfortran/lfortran#808]
```fortran
program s
   integer :: a(10)
   real(8) :: b(10)
   a = 2
   b = 0.2
   b(2) = 100.2
   a = int(b)
   print *, a(1:2)
end program
```
emits (intrinsics omitted, #807 applied)

```julia
function main()
    local a = Array{Int32, 1}(undef, 10)
    local b = Array{Float64, 1}(undef, 10)
    a .= 2
    b .=   2.00000000000000011e-01
    b[2] =   1.00200000000000003e+02
    a .= trunc(Int32, b)
    println(a[1:1:2])
end

main()
```

The problem is that `trunc(Int32, b)` does not work because `b` is an `Array` and broadcast needs to be used.

The current implementation tries to figure this out by calling `ASRUtils::is_array()` on `x.m_type` in `visit_Cast`

https://github.com/lfortran/lfortran/blob/4fb3fd0f8512451cf1d73368c0a77d9f8e11881f/src/libasr/codegen/asr_to_julia.cpp#L1458

But it seems that this did not work in the example above.

**Comment by lucifer1004** [comment]
We should check `x.m_arg->m_type` instead of `x.m_type`.

I have written a new version but am stuck on distinguishing whether `x.m_arg` is a variable or a constant.

**Comment by czgdp1807** [comment]
Cast for arrays is linear in the size of the array (a non-trivial operation). For LPython AFAIK, we plan to disallow such casting operations over arrays. For LFortran, it seems like converting `Cast` over arrays to element wise casting along with Broadcast usage is fine.

> We should check `x.m_arg->m_type` instead of `x.m_type`.

Correct.

**Comment by lucifer1004** [comment]
Fixed by #807
